diff --git a/meson.build b/meson.build
index 9153910e8a4..69d537361cd 100644
--- a/meson.build
+++ b/meson.build
@@ -813,7 +813,6 @@ if _opencl != 'disabled'
     error('The Clover OpenCL state tracker requires rtti')
   endif
 
-  with_clc = true
   with_gallium_opencl = true
   with_opencl_icd = _opencl == 'icd'
 else
@@ -837,11 +836,6 @@ if with_gallium_rusticl
   with_clc = true
 endif
 
-dep_clc = null_dep
-if with_clc
-  dep_clc = dependency('libclc')
-endif
-
 gl_pkgconfig_c_flags = []
 with_glx_indirect_rendering = false
 if with_platform_x11
@@ -1811,6 +1805,7 @@ pre_args += '-DLLVM_AVAILABLE=' + (with_llvm ? '1' : '0')
 pre_args += '-DDRAW_LLVM_AVAILABLE=' + (with_llvm and draw_with_llvm ? '1' : '0')
 
 with_opencl_spirv = (_opencl != 'disabled' and get_option('opencl-spirv')) or with_clc
+dep_clc = null_dep
 if with_opencl_spirv
   chosen_llvm_version_array = dep_llvm.version().split('.')
   chosen_llvm_version_major = chosen_llvm_version_array[0].to_int()
@@ -1833,6 +1828,7 @@ if with_opencl_spirv
     '>= @0@.@1@'.format(chosen_llvm_version_major, chosen_llvm_version_minor),
     '< @0@.@1@'.format(chosen_llvm_version_major, chosen_llvm_version_minor + 1) ]
 
+  dep_clc = dependency('libclc')
   dep_spirv_tools = dependency('SPIRV-Tools', required : true, version : '>= 2018.0')
   # LLVMSPIRVLib is available at https://github.com/KhronosGroup/SPIRV-LLVM-Translator
   dep_llvmspirvlib = dependency('LLVMSPIRVLib', required : true, version : _llvmspirvlib_version)
diff --git a/src/compiler/meson.build b/src/compiler/meson.build
index 1dae56d1b2b..8d73544c6d8 100644
--- a/src/compiler/meson.build
+++ b/src/compiler/meson.build
@@ -79,7 +79,7 @@ subdir('nir')
 
 subdir('spirv')
 
-if with_opencl_spirv
+if with_clc
   subdir('clc')
 endif
 if with_gallium
