Description: disable use of ppc64el assembly code
Author: Dmitry Shachnev <mitya57@debian.org>
Bug-Debian: https://bugs.debian.org/1092890
Last-Update: 2025-02-02

--- a/meson.build
+++ b/meson.build
@@ -1319,11 +1319,6 @@ elif host_machine.cpu_family() == 'sparc
     with_asm_arch = 'sparc'
     pre_args += ['-DUSE_SPARC_ASM']
   endif
-elif host_machine.cpu_family() == 'ppc64' and host_machine.endian() == 'little'
-  if system_has_kms_drm
-    with_asm_arch = 'ppc64le'
-    pre_args += ['-DUSE_PPC64LE_ASM']
-  endif
 elif host_machine.cpu_family() == 'mips64' and host_machine.endian() == 'little'
   if system_has_kms_drm
     with_asm_arch = 'mips64el'
