--- a/src/intel/vulkan/anv_device.c
+++ b/src/intel/vulkan/anv_device.c
@@ -1567,7 +1567,7 @@ VkResult anv_AllocateMemory(
        */
       alloc_flags |= ANV_BO_ALLOC_COMPRESSED;
    } else if (mem_type->compressed && !INTEL_DEBUG(DEBUG_NO_CCS)) {
-      alloc_flags |= ANV_BO_ALLOC_COMPRESSED;
+//      alloc_flags |= ANV_BO_ALLOC_COMPRESSED;
    }
 
    /* Anything imported or exported is EXTERNAL */
